### PR TITLE
Add KillWorker Plugin

### PR DESCRIPTION
**Pull request by mrocklin**
This is to be used for more strenuous stress testing

### CODE REVIEW

**Comment by mrocklin** [comment]
This is a stripped down version of #6123 both so that it can be merged more easily, and so that the other parts can be more easily reverted if necessary.

**Comment by mrocklin** [comment]
@gjoseph92 if you can do a quick review

**Review by gjoseph92 — APPROVED** [review]

**Comment by mrocklin** [comment]
Merging this on passed tests

File: distributed/tests/test_chaos.py [hunk 1]
```diff
@@ -4,10 +4,17 @@
...
+from distributed.utils_test import WINDOWS, gen_cluster
 
 
-@pytest.mark.parametrize("mode", ["sys.exit", "graceful", "segfault"])
+@pytest.mark.parametrize(
+    "mode",
+    [
+        "sys.exit",
+        "graceful",
+        pytest.param("segfault", marks=pytest.mark.skipif(WINDOWS, reason="unknown")),
```
Comment (gjoseph92) [inline distributed/tests/test_chaos.py:15]: Does windows... not segfault? Weird.

Comment (mrocklin) [reply distributed/tests/test_chaos.py:15]: Yeah, not sure.

Comment (mrocklin) [reply distributed/tests/test_chaos.py:15]: Or the ctypes trick doesn't operate the same on windows and would need to find some other better trick.

**Pull request merged**